larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -52,8 +54,24 @@ class LArMCParticle : public object_creation::MCParticle::Object
      */
     int GetNuanceCode() const;
 
+    /**
+     *  @brief  Get the position of the geant4 steps
+     *
+     *  @return vector of the positions
+     */
+    std::vector<pandora::CartesianVector> GetMCStepPositions() const;
+
+    /**
+     *  @brief  Get the momenta of the geant4 steps
+     *
+     *  @return vector of the momenta
+     */
+    std::vector<pandora::CartesianVector> GetMCStepMomentas() const;
```

Comment:
I'm wondering about the feasibility of these returning a const ref rather than returning by value (to avoid the potentially heavy cost of copying).  Unfortunately, things like RVO don't help here, I think.
Maybe ponder this one for a bit.

Reply:
I think a const reference makes sense here. The returned variable is a member variable, so there's no risk of returning a temporary and introducing compiler dependency for RVO, so the only question would be whether the returned vector might be edited at some point, at which time an explicit copy could be made as needed?